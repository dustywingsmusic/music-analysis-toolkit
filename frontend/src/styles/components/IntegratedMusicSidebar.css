/* Integrated Music Sidebar Styles */
.integrated-music-sidebar {
  position: fixed;
  top: 50px;
  right: 0;
  width: 320px;
  height: 91vh;
  background: #1e293b;
  border-left: 1px solid #475569;
  box-shadow: -4px 0 20px rgba(0, 0, 0, 0.3);
  z-index: 100;
  display: flex;
  flex-direction: column;
  overflow-y: auto;
  color: #e2e8f0;
}

/* Main content styling for reference tab - sidebar uses fixed positioning */
.reference-tab {
  /* Removed margin-right: 320px as sidebar uses fixed positioning and doesn't require content adjustment */
}

.reference-content {
  padding-right: 20px;
}

/* Sidebar Sections */
.sidebar-section {
  border-bottom: 1px solid #334155;
}

.sidebar-section:last-child {
  border-bottom: none;
  flex: 1;
}

/* Section Headers */
.sidebar-section-header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 12px 16px;
  background: #334155;
  cursor: pointer;
  transition: background-color 0.2s ease;
}

.sidebar-section-header:hover {
  background: #475569;
}

.sidebar-section-title {
  font-size: 0.875rem;
  font-weight: 600;
  margin: 0;
  color: #e2e8f0;
}

.sidebar-section-toggle {
  background: none;
  border: none;
  color: #94a3b8;
  font-size: 1rem;
  font-weight: bold;
  cursor: pointer;
  padding: 0;
  width: 20px;
  height: 20px;
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 2px;
  transition: all 0.2s ease;
}

.sidebar-section-toggle:hover {
  background: #475569;
  color: #e2e8f0;
}

/* Section Content */
.sidebar-section-content {
  padding: 16px;
  background: #1e293b;
}

/* Live Input Analysis Panel Integration */
.sidebar-midi-panel {
  margin: 0;
  padding: 0;
  background: transparent;
  border: none;
  box-shadow: none;
  /* Override fixed positioning from midi-detection-panel.css */
  position: static !important;
  top: auto !important;
  left: auto !important;
  transform: none !important;
  z-index: auto !important;
  width: 100% !important;
  max-width: none !important;
}

.sidebar-midi-panel .midi-detection-card {
  background: transparent;
  border: none;
  padding: 0;
  margin: 0;
}

/* Enhanced Live Input Analysis Styling for Sidebar */
.sidebar-midi-panel .midi-detection-card h3 {
  display: none; /* Hide the duplicate title since we have section header */
}

/* Detection Mode Styling */
.sidebar-midi-panel .mb-2 {
  margin-bottom: 12px;
}

.sidebar-midi-panel .text-xs {
  font-size: 0.75rem;
}

.sidebar-midi-panel .font-medium {
  font-weight: 500;
  color: #e2e8f0;
}

.sidebar-midi-panel .grid {
  display: grid;
  gap: 6px;
}

.sidebar-midi-panel .grid-cols-2 {
  grid-template-columns: repeat(2, 1fr);
}

.sidebar-midi-panel label {
  display: flex;
  align-items: center;
  gap: 6px;
  cursor: pointer;
  padding: 4px 6px;
  border-radius: 4px;
  transition: background-color 0.2s ease;
}

.sidebar-midi-panel label:hover {
  background: #334155;
}

.sidebar-midi-panel input[type="radio"] {
  width: 12px;
  height: 12px;
  accent-color: #06b6d4;
}

/* Chord Mode Settings Styling */
.sidebar-midi-panel .bg-slate-800 {
  background: #334155;
  border-radius: 6px;
  padding: 12px;
  margin-bottom: 12px;
}

.sidebar-midi-panel select {
  width: 100%;
  padding: 6px 8px;
  font-size: 0.75rem;
  background: #1e293b;
  border: 1px solid #475569;
  border-radius: 4px;
  color: #e2e8f0;
  transition: border-color 0.2s ease;
}

.sidebar-midi-panel select:focus {
  outline: none;
  border-color: #06b6d4;
}

.sidebar-midi-panel select option {
  background: #1e293b;
  color: #e2e8f0;
}

/* Notes Display Styling */
.sidebar-midi-panel .min-h-\[18px\] {
  min-height: 18px;
  padding: 8px;
  background: #334155;
  border-radius: 4px;
  font-family: 'Courier New', monospace;
  font-size: 0.75rem;
  color: #e2e8f0;
  border: 1px solid #475569;
}

.sidebar-midi-panel button {
  padding: 4px 8px;
  font-size: 0.75rem;
  background: #475569;
  color: #e2e8f0;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  transition: background-color 0.2s ease;
}

.sidebar-midi-panel button:hover {
  background: #64748b;
}

/* Responsive adjustments for MIDI panel */
@media (max-width: 768px) {
  .sidebar-midi-panel .grid-cols-2 {
    grid-template-columns: 1fr;
  }

  .sidebar-midi-panel .bg-slate-800 {
    padding: 8px;
  }
}

/* Temporary highlight effect for scales clicked from sidebar */
.sidebar-highlighted {
  background: rgba(6, 182, 212, 0.3) !important;
  border: 2px solid #06b6d4 !important;
  box-shadow: 0 0 10px rgba(6, 182, 212, 0.5) !important;
  transition: all 0.3s ease !important;
}

/* Status indicators */
.sidebar-status-indicator {
  display: inline-block;
  width: 8px;
  height: 8px;
  border-radius: 50%;
  margin-right: 6px;
}

.sidebar-status-indicator.active {
  background: #10b981;
  box-shadow: 0 0 4px rgba(16, 185, 129, 0.5);
}

.sidebar-status-indicator.inactive {
  background: #64748b;
}

.sidebar-status-indicator.detecting {
  background: #f59e0b;
  animation: pulse 1.5s infinite;
}

@keyframes pulse {
  0%, 100% {
    opacity: 1;
  }
  50% {
    opacity: 0.5;
  }
}

/* Suggestions Styling */
.melody-suggestions,
.chord-suggestions {
  display: flex;
  flex-direction: column;
  gap: 12px;
}

.suggestion-item {
  background: #334155;
  border-radius: 6px;
  padding: 12px;
  border-left: 4px solid #06b6d4;
  transition: all 0.2s ease;
}

.suggestion-item:hover {
  background: #475569;
  transform: translateX(2px);
}

.suggestion-header {
  font-weight: 600;
  color: #e2e8f0;
  margin-bottom: 4px;
  font-size: 0.875rem;
}

.suggestion-confidence {
  color: #94a3b8;
  font-size: 0.75rem;
  margin-bottom: 8px;
}

.suggestion-scales {
  margin-top: 8px;
  padding-top: 8px;
  border-top: 1px solid #475569;
}

.suggestion-scales small {
  display: block;
  color: #64748b;
  margin-bottom: 4px;
  font-size: 0.75rem;
}

.scale-link {
  color: #06b6d4;
  cursor: pointer;
  text-decoration: underline;
  font-size: 0.75rem;
  transition: color 0.2s ease;
}

.scale-link:hover {
  color: #0891b2;
}

/* No suggestions/results states */
.no-suggestions,
.no-results {
  color: #64748b;
  text-align: center;
  padding: 20px;
  font-style: italic;
  font-size: 0.875rem;
}

/* Analysis Results */
.analysis-results {
  color: #e2e8f0;
}

.analysis-placeholder {
  color: #94a3b8;
  font-style: italic;
  text-align: center;
  padding: 20px;
  font-size: 0.875rem;
}

/* Enhanced Analysis Results Styling */
.analysis-loading {
  text-align: center;
  padding: 20px;
  color: #94a3b8;
}

.loading-indicator {
  font-size: 1.5rem;
  animation: pulse 1.5s infinite;
  margin-bottom: 8px;
}

.analysis-error {
  background: #7f1d1d;
  border: 1px solid #ef4444;
  border-radius: 6px;
  padding: 12px;
  margin-bottom: 12px;
}

.analysis-error h5 {
  color: #fca5a5;
  margin: 0 0 8px 0;
  font-size: 0.875rem;
}

.analysis-error p {
  color: #fecaca;
  margin: 0;
  font-size: 0.75rem;
}

.analysis-primary {
  background: #334155;
  border-radius: 6px;
  padding: 12px;
  margin-bottom: 12px;
  border-left: 4px solid #10b981;
}

.analysis-primary h5 {
  color: #10b981;
  margin: 0 0 8px 0;
  font-size: 0.875rem;
}

.mode-result {
  margin-bottom: 8px;
}

.mode-result strong {
  color: #e2e8f0;
  font-size: 1rem;
}

.confidence {
  color: #94a3b8;
  font-size: 0.75rem;
  margin-left: 8px;
}

.scale-notes {
  margin-bottom: 12px;
}

.scale-notes small {
  color: #94a3b8;
  font-size: 0.75rem;
}

.view-in-tables-btn {
  background: #10b981;
  color: white;
  border: none;
  padding: 6px 12px;
  border-radius: 4px;
  font-size: 0.75rem;
  cursor: pointer;
  transition: background-color 0.2s ease;
}

.view-in-tables-btn:hover {
  background: #059669;
}

.analysis-explanation {
  background: #1e293b;
  border-radius: 6px;
  padding: 12px;
  margin-top: 12px;
  border-left: 4px solid #06b6d4;
}

.analysis-explanation h6 {
  color: #06b6d4;
  margin: 0 0 8px 0;
  font-size: 0.75rem;
  font-weight: 600;
}

.analysis-explanation p {
  color: #cbd5e1;
  margin: 0;
  font-size: 0.75rem;
  line-height: 1.4;
}

.analysis-local {
  background: #334155;
  border-radius: 6px;
  padding: 12px;
  border-left: 4px solid #f59e0b;
}

.analysis-local h5 {
  color: #f59e0b;
  margin: 0 0 8px 0;
  font-size: 0.875rem;
}

.local-suggestions {
  display: flex;
  flex-direction: column;
  gap: 6px;
  margin-bottom: 8px;
}

.local-suggestion {
  display: flex;
  align-items: center;
  gap: 8px;
}

.local-suggestion strong {
  color: #e2e8f0;
  font-size: 0.75rem;
}

.played-notes small {
  color: #94a3b8;
  font-size: 0.75rem;
}

/* Responsive Design */
@media (max-width: 1024px) {
  .integrated-music-sidebar {
    width: 280px;
  }
}

@media (max-width: 768px) {
  .integrated-music-sidebar {
    position: fixed;
    top: auto;
    bottom: 0;
    left: 0;
    right: 0;
    width: 100%;
    height: 50vh;
    max-height: 400px;
    border-left: none;
    border-top: 1px solid #475569;
    box-shadow: 0 -4px 20px rgba(0, 0, 0, 0.3);
  }

  .reference-tab {
    /* Mobile styles removed - sidebar uses fixed positioning */
  }

  .sidebar-section-content {
    padding: 12px;
  }

  .suggestion-item {
    padding: 10px;
  }
}

/* Scrollbar Styling */
.integrated-music-sidebar::-webkit-scrollbar {
  width: 6px;
}

.integrated-music-sidebar::-webkit-scrollbar-track {
  background: #1e293b;
}

.integrated-music-sidebar::-webkit-scrollbar-thumb {
  background: #475569;
  border-radius: 3px;
}

.integrated-music-sidebar::-webkit-scrollbar-thumb:hover {
  background: #64748b;
}

/* Animation for section expansion */
.sidebar-section-content {
  animation: slideDown 0.2s ease-out;
}

@keyframes slideDown {
  from {
    opacity: 0;
    transform: translateY(-10px);
  }
  to {
    opacity: 1;
    transform: translateY(0);
  }
}

/* Focus states for accessibility */
.sidebar-section-header:focus,
.sidebar-section-toggle:focus,
.scale-link:focus {
  outline: 2px solid #06b6d4;
  outline-offset: 2px;
}

/* High contrast mode support */
@media (prefers-contrast: high) {
  .integrated-music-sidebar {
    border-left: 2px solid #ffffff;
  }

  .sidebar-section {
    border-bottom: 2px solid #ffffff;
  }

  .suggestion-item {
    border-left: 4px solid #ffffff;
  }
}

/* Reduced motion support */
@media (prefers-reduced-motion: reduce) {
  .suggestion-item,
  .sidebar-section-toggle,
  .scale-link {
    transition: none;
  }

  .sidebar-section-content {
    animation: none;
  }
}

/* Progressive Disclosure Controls */
.progressive-disclosure-controls {
  margin-top: 16px;
  padding-top: 12px;
  border-top: 1px solid #475569;
  display: flex;
  flex-direction: column;
  gap: 8px;
}

.show-more-btn,
.toggle-analysis-btn,
.toggle-options-btn {
  background: #475569;
  color: #e2e8f0;
  border: none;
  padding: 8px 12px;
  border-radius: 6px;
  font-size: 0.75rem;
  cursor: pointer;
  transition: all 0.2s ease;
  display: flex;
  align-items: center;
  gap: 4px;
}

.show-more-btn:hover,
.toggle-analysis-btn:hover,
.toggle-options-btn:hover {
  background: #64748b;
  transform: translateY(-1px);
}

.show-more-btn:focus,
.toggle-analysis-btn:focus,
.toggle-options-btn:focus {
  outline: 2px solid #06b6d4;
  outline-offset: 2px;
}

.advanced-options {
  background: #1e293b;
  border-radius: 6px;
  padding: 12px;
  margin-top: 8px;
  border: 1px solid #334155;
  animation: slideDown 0.2s ease-out;
}

.advanced-options .option-label {
  display: block;
  color: #cbd5e1;
  font-size: 0.75rem;
  margin-bottom: 8px;
}

.advanced-options input[type="range"] {
  width: 100%;
  margin-top: 4px;
}

/* Real-time Mode Detection Styles */
.mode-detection-results {
  padding: 12px;
}

.mode-state-indicator {
  margin-bottom: 12px;
  text-align: center;
}

.mode-badge {
  display: inline-block;
  padding: 6px 12px;
  border-radius: 16px;
  font-size: 0.75rem;
  font-weight: 600;
  text-transform: uppercase;
  letter-spacing: 0.5px;
}

.mode-badge.scale-mode {
  background: #065f46;
  color: #d1fae5;
  border: 1px solid #10b981;
}

.mode-badge.melody-mode {
  background: #7c2d12;
  color: #fed7aa;
  border: 1px solid #f97316;
}

/* Manual Override Controls */
.mode-controls {
  display: flex;
  gap: 8px;
  margin-bottom: 16px;
  justify-content: center;
}

.continue-scale-mode-btn,
.enter-melody-mode-btn {
  background: #475569;
  color: #e2e8f0;
  border: none;
  padding: 8px 12px;
  border-radius: 6px;
  font-size: 0.75rem;
  cursor: pointer;
  transition: all 0.2s ease;
  font-weight: 500;
}

.continue-scale-mode-btn:hover {
  background: #10b981;
  transform: translateY(-1px);
}

.enter-melody-mode-btn:hover {
  background: #f97316;
  transform: translateY(-1px);
}

.continue-scale-mode-btn:focus,
.enter-melody-mode-btn:focus {
  outline: 2px solid #06b6d4;
  outline-offset: 2px;
}

/* Root Picker */
.root-picker {
  background: #1e293b;
  border-radius: 8px;
  padding: 12px;
  margin-bottom: 16px;
  border: 1px solid #334155;
}

.root-picker h6 {
  margin: 0 0 8px 0;
  color: #cbd5e1;
  font-size: 0.75rem;
  text-align: center;
}

.root-picker-buttons {
  display: grid;
  grid-template-columns: repeat(6, 1fr);
  gap: 4px;
}

.root-picker-btn {
  background: #334155;
  color: #e2e8f0;
  border: none;
  padding: 8px 4px;
  border-radius: 4px;
  font-size: 0.75rem;
  cursor: pointer;
  transition: all 0.2s ease;
  font-weight: 500;
}

.root-picker-btn:hover {
  background: #475569;
  transform: translateY(-1px);
}

.root-picker-btn:focus {
  outline: 2px solid #06b6d4;
  outline-offset: 2px;
}

/* Melody Mode Message */
.melody-mode-message {
  background: #7c2d12;
  border: 1px solid #f97316;
  border-radius: 6px;
  padding: 12px;
  margin-bottom: 16px;
}

.melody-mode-message p {
  margin: 0 0 8px 0;
  color: #fed7aa;
  font-size: 0.75rem;
  line-height: 1.4;
}

.melody-mode-message p:last-child {
  margin-bottom: 0;
}

/* Grouped Suggestions */
.grouped-suggestions {
  margin-top: 16px;
}

.family-groups {
  display: flex;
  flex-direction: column;
  gap: 8px;
}

.family-title {
  margin: 0;
  font-size: 0.75rem;
  font-weight: 600;
  color: #e2e8f0;
  display: flex;
  align-items: center;
  gap: 6px;
}

.family-count {
  font-size: 0.625rem;
  color: #94a3b8;
  font-weight: 400;
}

.family-suggestions {
  background: #1e293b;
  padding: 8px;
}

.family-suggestions .suggestion-item {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 6px 8px;
  margin-bottom: 4px;
  background: #0f172a;
  border-radius: 4px;
  border: 1px solid #334155;
}

.family-suggestions .suggestion-item:last-child {
  margin-bottom: 0;
}

.suggestion-header {
  display: flex;
  flex-direction: column;
  gap: 2px;
  flex: 1;
}

.suggestion-name {
  font-size: 0.75rem;
  color: #e2e8f0;
  font-weight: 500;
}

.suggestion-mismatch {
  font-size: 0.625rem;
  color: #94a3b8;
}

.preview-btn {
  background: none;
  border: none;
  color: #94a3b8;
  cursor: pointer;
  padding: 4px;
  border-radius: 4px;
  transition: all 0.2s ease;
  font-size: 0.875rem;
}

.preview-btn:hover {
  background: #334155;
  color: #e2e8f0;
  transform: scale(1.1);
}

.preview-btn:focus {
  outline: 2px solid #06b6d4;
  outline-offset: 2px;
}

/* Mode Detection Section */
.mode-detection-section {
  margin-bottom: 16px;
}

.mode-detection-section h5 {
  margin: 0 0 12px 0;
  font-size: 0.875rem;
  font-weight: 600;
  color: #cbd5e1;
}

