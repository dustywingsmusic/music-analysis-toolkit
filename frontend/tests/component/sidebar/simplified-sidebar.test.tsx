import { render, fireEvent } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import IntegratedMusicSidebar from '@/components/IntegratedMusicSidebar';

/** Snapshot tests for minimal sidebar rendering */

describe('IntegratedMusicSidebar snapshots', () => {
  it('renders collapsed sidebar', () => {
    const { container } = render(<IntegratedMusicSidebar />);
    expect(container).toMatchSnapshot();
  });

  it('renders expanded empty sidebar', () => {
    const { container, getByLabelText } = render(<IntegratedMusicSidebar />);
    fireEvent.click(getByLabelText('Show music analysis sidebar'));
    expect(container).toMatchSnapshot();
  });
});
