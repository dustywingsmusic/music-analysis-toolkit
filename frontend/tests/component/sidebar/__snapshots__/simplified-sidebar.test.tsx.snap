// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IntegratedMusicSidebar snapshots > renders collapsed sidebar 1`] = `
<div>
  <button
    aria-label="Show music analysis sidebar"
    class="sidebar-toggle-btn"
    style="position: fixed; right: 20px; top: 50%; transform: translateY(-50%); z-index: 1001; background: rgb(71, 85, 105); color: rgb(226, 232, 240); border-radius: 8px 0 0 8px; padding: 12px 8px; cursor: pointer; transition: all 0.3s ease; font-size: 1.2rem; box-shadow: 0 2px 8px rgba(0, 0, 0, 0.3);"
  >
    ðŸŽµ
  </button>
</div>
`;

exports[`IntegratedMusicSidebar snapshots > renders expanded empty sidebar 1`] = `
<div>
  <button
    aria-label="Hide music analysis sidebar"
    class="sidebar-toggle-btn"
    style="position: fixed; right: 320px; top: 50%; transform: translateY(-50%); z-index: 1001; background: rgb(71, 85, 105); color: rgb(226, 232, 240); border-radius: 8px 0 0 8px; padding: 12px 8px; cursor: pointer; transition: all 0.3s ease; font-size: 1.2rem; box-shadow: 0 2px 8px rgba(0, 0, 0, 0.3);"
  >
    â†’
  </button>
  <div
    class="integrated-music-sidebar "
  >
    <div
      class="sidebar-section"
    >
      <div
        class="sidebar-section-header"
      >
        <h3
          class="sidebar-section-title"
        >
          <span
            class="sidebar-status-indicator inactive"
          />
          ðŸŽ¹ Live Input Analysis
        </h3>
        <button
          class="sidebar-section-toggle"
        >
          âˆ’
        </button>
      </div>
      <div
        class="sidebar-section-content"
      >
        <div
          class="midi-detection-panel sidebar-midi-panel"
        >
          <div
            class="midi-detection-card"
          >
            <h3
              class="text-xs font-semibold mb-2 text-cyan-400"
            >
              Live Input Analysis
            </h3>
            <div
              class="mb-0"
            >
              <div
                class="flex justify-between items-center mb-1"
              >
                <p
                  class="text-xs font-medium"
                >
                  Notes Detected:
                </p>
                <div
                  class="flex gap-1"
                >
                  <button
                    class="px-1.5 py-0.5 text-xs bg-slate-600 hover:bg-slate-500 rounded transition-colors"
                  >
                    Clear
                  </button>
                </div>
              </div>
              <div
                class="min-h-[18px] p-1.5 bg-slate-800 rounded text-xs font-mono flex flex-wrap gap-1 items-center"
              >
                No notes detected
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sidebar-section"
    >
      <div
        class="sidebar-section-header"
      >
        <h3
          class="sidebar-section-title"
        >
          <span
            class="sidebar-status-indicator inactive"
          />
          ðŸŽ¯ Musical Analysis
        </h3>
        <button
          class="sidebar-section-toggle"
        >
          âˆ’
        </button>
      </div>
      <div
        class="sidebar-section-content"
      >
        <div
          class="no-results"
        >
          Play some notes to see analysis
        </div>
      </div>
    </div>
  </div>
</div>
`;
